documentation/projects/proposals/document_all_media_properties/20230426-implementation_plan_documenting_media_properties.md 

```diff
@@ -0,0 +1,82 @@
+
+- As a maintainer/contributor, I need to know the list of available properties for each media type in the catalog database. ... so that?
+- As a contributor, I need to know what is the expected data type for the specific property. Is it required? ... so that?
+- As a contributor, I need to know how the data property is transformed and validated.
+- As a contributor writing a provider API script, which data properties do I need to collect?
+- As a contributor writing a provider API script, how do I select data for the specific media property Openverse uses?
+- As a contributor writing a provider script, which format should I convert a media property?
+- As a maintainer, I've found a discrepancy between the type of data property the API returns and what is described in the table (e.g., is `null` for a non-nullable property). Is there an issue to fix this data in the catalog database, or do I need to create a new one?
+- As an API user, I need to know what exactly this media property is and what data points from the providers are used for it. E.g., does the "creator" refer to the person who created the object or took the picture?
+- As a provider of openly-licensed media, I want to know what media properties Openverse collects so that I can change my API responses.
+
+The information on data types, transformations, and validations will be presented as a table and will be extracted from the code.  Other information about the data selection properties and data discrepancies will be written in a separate markdown file as it is easier to read, write, and lint.
+
+The catalog has a DAG documentation generator that extracts the docstrings from the DAGs and creates a page with the documentation. We can use the same approach to create a documentation page for the media properties. The DAG doc generator runs the code inside docker to extract DAG information. In this project, we can simplify and use the `ast` module to parse static code and extract the docstrings to make the checks faster. This will, however, mean that some of the settings will not be picked up and will have to be updated during the generation. One example of this is the `nullable` field in the `Column` class: it falls back to the value of `not nullable` during the object initiation.
+
+## Outlined Steps
+
+### `just` script
+
+The script parses the SQL DDL files, Python files and markdown files to generate a markdown page with a table and long-form notes. This script can `diff` with the existing documentation page (created in step 4 below) for the `precommit` check and post a note about how to update the docs if differences are detected.
+
+### 1. SQL DDL parsing
+We will use the names of the properties as they are in the upstream database, in the same order as they are in the `image_view`/`audio_view` materialized views. These views are selected over the main `image` and `audio` tables because they are used by the API during data refresh and contain a complete list of properties.
```

Comment:
I've added a note about the #1894 changes, but forgot to push it :) It would actually make generating the docs much simpler - we would need to only parse one sql file per media type.

Reply:
+1 to linking the DDL files in question here!